File: XPile 3-17/Programs/test_list.js

function main() {
    let scores = [];
    scores.push(10);
    scores.push(20);
    scores.push(30);
    console.log(scores.length);
    console.log(scores[0]);
    console.log(scores[1]);
    console.log(scores[2]);
    (scores.splice(1, 0, 15), undefined);
    console.log(scores[1]);
    console.log(scores.length);
    scores.splice(scores.indexOf(15), 1);
    console.log(scores.length);
    let dropped = scores.splice(0, 1)[0];
    console.log(dropped);
    console.log(scores.length);
    let nums = [5, 3, 1, 4, 2];
    console.log(nums.length);
    nums.sort((a, b) => a < b ? -1 : a > b ? 1 : 0);
    console.log(nums[0]);
    console.log(nums[4]);
    nums.reverse();
    console.log(nums[0]);
    console.log(nums[4]);
    if (nums.includes(3)) {
        console.log("found 3");
    }
    (nums.length = 0);
    console.log(nums.length);
    let items = [100, 200, 300];
    let total = 0;
    for (const x of items) {
        total = (total + x);
    }
    console.log(total);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_list.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    scores = []
    scores.append(10)
    scores.append(20)
    scores.append(30)
    print(len(scores))
    print(scores[0])
    print(scores[1])
    print(scores[2])
    scores.insert(1, 15)
    print(scores[1])
    print(len(scores))
    scores.remove(15)
    print(len(scores))
    dropped = scores.pop(0)
    print(dropped)
    print(len(scores))
    nums = [5, 3, 1, 4, 2]
    print(len(nums))
    nums.sort()
    print(nums[0])
    print(nums[4])
    nums.reverse()
    print(nums[0])
    print(nums[4])
    if (3 in nums):
        print('found 3')
    nums.clear()
    print(len(nums))
    items = [100, 200, 300]
    total = 0
    for x in items:
        total = (total + x)
    print(total)
    return 0


if __name__ == "__main__":
    main()
